pkg/cluster/delete.go

```diff
@@ -41,12 +48,32 @@ func Delete(
 			})
 		return err
 	}
+
 	clusterQuery := "arlon-cluster=" + name
 	apps, err := appIf.List(context.Background(),
 		&argoapp.ApplicationQuery{Selector: &clusterQuery})
 	if err != nil {
 		return fmt.Errorf("failed to list apps related to cluster: %s", err)
 	}
+
+	app, err := appIf.Get(context.Background(),
```

Comment:
It's unnecessary to get the "cluster-app" app here, because line 53 already gets all apps related to this cluster, and lines 77-92 iterate over those apps. So you should insert code in that loop to catch the one of type cluster-app, and then handle it.

Reply:
removed the unnecessary calling of appIf.List again and used the loop in lines 77-92 to do the same